pkg/backup: BenchmarkRestoreEntryCover failed

pkg/backup.BenchmarkRestoreEntryCover [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/21324847?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/21324847?buildTab=artifacts#./artifacts/microbench/experiment) on master @ [6f656da91f55633e0309ece9b568f93df1cf0391](https://github.com/cockroachdb/cockroach/commits/6f656da91f55633e0309ece9b568f93df1cf0391):

Fatal error:

```
panic: nodelocal test implementation in non-test build

```
Stack: 

```
goroutine 52 gp=0x1c7249f052c0 m=27 mp=0x1c7249401808 [running]:
panic({0x61ea2e0?, 0x74e8960?})
	GOROOT/src/runtime/panic.go:879 +0x16f fp=0x1c724bbcf6a0 sp=0x1c724bbcf5f0 pc=0x4c83af
github.com/cockroachdb/cockroach/pkg/cloud/nodelocal.TestingMakeNodelocalStorage(...)
	pkg/cloud/nodelocal/test_nodelocal_storage.go:55
github.com/cockroachdb/cockroach/pkg/backup.BenchmarkRestoreEntryCover.func1({0x0?, 0x0?}, {0x1, {{0x1c724b056080, 0x3c}, 0x0}, {{0x0, 0x0}}, 0x0, 0x0, ...}, ...)
	pkg/backup/bench_covering_test.go:67 +0x25 fp=0x1c724bbcf6c0 sp=0x1c724bbcf6a0 pc=0x5a82565
github.com/cockroachdb/cockroach/pkg/backup.MockBackupChain({0x755e9d0, 0xcc39100}, 0x1, 0xa, 0x0, 0x100000, 0x1c724a56e660, 0x1, 0x1c724ac06080)
	pkg/backup/restore_span_covering_test.go:119 +0xe12 fp=0x1c724bbcfcd8 sp=0x1c724bbcf6c0 pc=0x5afcb72
github.com/cockroachdb/cockroach/pkg/backup.BenchmarkRestoreEntryCover.func2.1.1.1(0x1c7249f5c608)
	pkg/backup/bench_covering_test.go:82 +0x6b fp=0x1c724bbcff10 sp=0x1c724bbcfcd8 pc=0x5a8184b
testing.(*B).runN(0x1c7249f5c608, 0x1)
	GOROOT/src/testing/benchmark.go:219 +0x190 fp=0x1c724bbcffa0 sp=0x1c724bbcff10 pc=0x5bf770
testing.(*B).run1.func1()
	GOROOT/src/testing/benchmark.go:245 +0x48 fp=0x1c724bbcffe0 sp=0x1c724bbcffa0 pc=0x5bfda8
runtime.goexit({})
	src/runtime/asm_amd64.s:1771 +0x1 fp=0x1c724bbcffe8 sp=0x1c724bbcffe0 pc=0x4d1301
created by testing.(*B).run1 in goroutine 51
	GOROOT/src/testing/benchmark.go:238 +0x90
```
<details><summary>Log preceding fatal error</summary>
<p>

```
goos: linux
goarch: amd64
cpu: Intel(R) Xeon(R) CPU @ 2.60GHz
BenchmarkRestoreEntryCover
BenchmarkRestoreEntryCover/numBackups=1
BenchmarkRestoreEntryCover/numBackups=1/numFiles=0
BenchmarkRestoreEntryCover/numBackups=1/numFiles=0/numSpans=10
BenchmarkRestoreEntryCover/numBackups=1/numFiles=0/numSpans=10/hasExternalFilesList=true

I260423 21:57:43.746040 1 (gostd) rand.go:288  [-] 1  random seed: -4701923800028719863
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*BenchmarkRestoreEntryCover.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63214

--- a/pkg/backup/bench_covering_test.go
+++ b/pkg/backup/bench_covering_test.go
@@ -11,11 +11,8 @@ import (
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/backup/backupinfo"
-	"github.com/cockroachdb/cockroach/pkg/cloud"
-	"github.com/cockroachdb/cockroach/pkg/cloud/cloudpb"
-	"github.com/cockroachdb/cockroach/pkg/cloud/nodelocal"
 	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
-	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
@@ -24,11 +21,10 @@ import (
 )
 
 func BenchmarkCoverageChecks(b *testing.B) {
-	dir := b.TempDir()
-	settings := cluster.MakeTestingClusterSettings()
-	storageFactory := func(ctx context.Context, dest cloudpb.ExternalStorage, opts ...cloud.ExternalStorageOption) (cloud.ExternalStorage, error) {
-		return nodelocal.TestingMakeNodelocalStorage(dir, settings, dest), nil
-	}
+	tc, _, _, cleanupFn := backupRestoreTestSetup(b, singleNode, 2, InitManualReplication)
+	defer cleanupFn()
+	execCfg := tc.ApplicationLayer(0).ExecutorConfig().(sql.ExecutorConfig)
+	storageFactory := execCfg.DistSQLSrv.ExternalStorage
 	ctx := context.Background()
 	r, _ := randutil.NewTestRand()
 
@@ -61,11 +57,10 @@ func BenchmarkCoverageChecks(b *testing.B) {
 }
 
 func BenchmarkRestoreEntryCover(b *testing.B) {
-	dir := b.TempDir()
-	settings := cluster.MakeTestingClusterSettings()
-	storageFactory := func(ctx context.Context, dest cloudpb.ExternalStorage, opts ...cloud.ExternalStorageOption) (cloud.ExternalStorage, error) {
-		return nodelocal.TestingMakeNodelocalStorage(dir, settings, dest), nil
-	}
+	tc, _, _, cleanupFn := backupRestoreTestSetup(b, singleNode, 2, InitManualReplication)
+	defer cleanupFn()
+	execCfg := tc.ApplicationLayer(0).ExecutorConfig().(sql.ExecutorConfig)
+	storageFactory := execCfg.DistSQLSrv.ExternalStorage
 
 	ctx := context.Background()
 	r, _ := randutil.NewTestRand()

